File: frontend/src/api/kernel.js
import axios from 'axios'

const api = axios.create({
  baseURL: '/api',
  timeout: 10000
})

// Kernel API
export const kernelApi = {
  // Get current kernel status
  getStatus() {
    return api.get('/kernel/status')
  },

  // Get system info
  getSystemInfo() {
    return api.get('/kernel/system')
  },

  // Get Go runtime monitor stats
  getMonitor() {
    return api.get('/kernel/monitor')
  }
}

export default api

File: backend/handlers/kernel.go
package handlers

import (
	"net/http"

	"sing-panel/services"

	"github.com/gin-gonic/gin"
)

type KernelHandler struct {
	service *services.KernelService
}

func NewKernelHandler(service *services.KernelService) *KernelHandler {
	return &KernelHandler{service: service}
}

// GetStatus returns the current kernel status
func (h *KernelHandler) GetStatus(c *gin.Context) {
	status := h.service.GetStatus()
	c.JSON(http.StatusOK, gin.H{
		"success": true,
		"data":    status,
	})
}

// GetSystemInfo returns system platform and architecture info
func (h *KernelHandler) GetSystemInfo(c *gin.Context) {
	info := h.service.GetSystemInfo()
	c.JSON(http.StatusOK, gin.H{
		"success": true,
		"data":    info,
	})
}

// GetMonitor returns Go runtime performance metrics
func (h *KernelHandler) GetMonitor(c *gin.Context) {
	stats := h.service.GetMonitorStats()
	c.JSON(http.StatusOK, gin.H{
		"success": true,
		"data":    stats,
	})
}
